platform/ext/target/nordic_nrf/common/plat_test.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tfm_plat_test.h"
+#include <stdint.h>
+#include <stdbool.h>
+#include <hal/nrf_gpio.h>
+#include <hal/nrf_timer.h>
+#include <nrf_board.h>
+
+#define TIMER_RELOAD_VALUE (16*1024*1024)
+#define USERLED_MASK       (1UL)
+
+static bool initialized = false;
+
+static void gpio_init(void)
+{
+    nrf_gpio_cfg(BUTTON1_PIN, NRF_GPIO_PIN_DIR_INPUT,
+                NRF_GPIO_PIN_INPUT_DISCONNECT, BUTTON1_PULL,
+                NRF_GPIO_PIN_S0S1, NRF_GPIO_PIN_NOSENSE);
+    nrf_gpio_cfg(LED1_PIN, NRF_GPIO_PIN_DIR_INPUT,
+                NRF_GPIO_PIN_INPUT_DISCONNECT, NRF_GPIO_PIN_NOPULL,
+                NRF_GPIO_PIN_S0S1, NRF_GPIO_PIN_NOSENSE);
+    initialized = true;
+}
+
+void tfm_plat_test_wait_user_button_pressed(void)
+{
+    if (!initialized) gpio_init();
+
+    /* Wait until button is pressed */
+    while (nrf_gpio_pin_read(BUTTON1_PIN) != BUTTON1_ACTIVE_HIGH) { ; }
+}
+
+void tfm_plat_test_wait_user_button_released(void)
+{
+    if (!initialized) gpio_init();
+
+    /* Wait until user button 0 is released */
+    while (nrf_gpio_pin_read(BUTTON1_PIN) == BUTTON1_ACTIVE_HIGH) { ; }
+}
+
+uint32_t tfm_plat_test_get_led_status(void)
+{
+    if (!initialized) gpio_init();
+
+    return (nrf_gpio_pin_out_read(LED1_PIN) == LED1_ACTIVE_HIGH);
+}
+
+void tfm_plat_test_set_led_status(uint32_t status)
+{
+    if (!initialized) gpio_init();
+
+    nrf_gpio_pin_write(LED1_PIN, status == LED1_ACTIVE_HIGH);
+}
+
+uint32_t tfm_plat_test_get_userled_mask(void)
+{
+    return USERLED_MASK;
+}
+
+static void timer_start(NRF_TIMER_Type * TIMER)
+{
+    nrf_timer_mode_set(TIMER, NRF_TIMER_MODE_TIMER);
+    nrf_timer_bit_width_set(TIMER, NRF_TIMER_BIT_WIDTH_24);
+    nrf_timer_frequency_set(TIMER, NRF_TIMER_FREQ_16MHz);
+    nrf_timer_cc_set(TIMER, NRF_TIMER_CC_CHANNEL0, TIMER_RELOAD_VALUE);
```

Comment:
As the timer is configured to use 24 bits, this will effectively set the compare value to 0 (the reload value is 2^24 currently). It will work, but in case the reload value is increased a bit, the effect might be be surprising.
Why not use the timer in 32-bit mode?

Reply:
I remember at some point not all timers supported 32 bit, but I found nothing like that for 91. Changing to 32bit.